src/firmware/app/control/trajectory_planner.h

```diff
@@ -0,0 +1,147 @@
+#pragma once
+
+#include "firmware/shared/math/polynomial_2d.h"
+#include "firmware/shared/math/vector_2d.h"
+
+#define TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS                                              \
+    6000  // The maximum size of the array containing trajectory elements
+
+// Struct that defines a single point on a trajectory
+// Includes the Position,and Time data corresponding to that point
+typedef struct TrajectoryElement
+{
+    Vector2d_t position;
+    float time;
+} TrajectoryElement_t;
+
+typedef struct Trajectory
+{
+    TrajectoryElement_t* trajectory_elements;
+    unsigned int num_elements;
+} Trajectory_t;
+
+/*
+ *   t_start - The path parameter value indicating the beginning of the
+ *       considered path
+ *   t_end - The path parameter value indicating the end of the considered
+ *       path
+ *
+ *   num_segments - The number of segments to discretize the trajectory into.
+ *       Must be greater than 2 segments.
+ *       THE NUMBER OF SEGMENTS MUST BE UNDER
+ *       TRAJECTORY_PLANNER_MAX_NUMBER_SEGMENTS
+ *
+ *   max_allowable_acceleration - The maximum acceleration allowed at any
+ *       point along the trajectory. This factor limits the maximum delta-velocity and
+ * also the max speed around curves due to centripetal acceleration [m/s^2]
+ *
+ *   max_allowable_speed - The maximum speed allowable at any point along the
+ *
+ *   path [m/s] *initial_speed - The initial speed at the start of the trajectory [m/s]
+ *
+ *   final_speed - The final speed at the end of the trajectory [m/s]
+ */
+typedef struct FirmwareRobotPathParameters
+{
+    Polynomial2dOrder3_t path;
+    float t_start;
+    float t_end;
+    unsigned int num_segments;
+    float max_allowable_acceleration;
+    float max_allowable_speed;
+    float initial_speed;
+    float final_speed;
+} FirmwareRobotPathParameters_t;
+
+/**
+ * Returns a planned trajectory as a list of TrajectoryElement's.
```

Comment:
``` suggestion
 * Returns a planned trajectory.
```

Reply:
Changed to:
Returns a planned trajectory with the list of guarantees based on the assumptions below